src/cpp/src/image_generation/models/autoencoder_kl.cpp

```diff
@@ -182,7 +182,19 @@ AutoencoderKL::AutoencoderKL(const std::string& vae_encoder_model,
     compile(device, *extract_adapters_from_properties(properties));
 }
 
-AutoencoderKL::AutoencoderKL(const AutoencoderKL&) = default;
+AutoencoderKL::AutoencoderKL(const AutoencoderKL& rhs) = default;
+
+AutoencoderKL AutoencoderKL::clone() {
+    AutoencoderKL cloned = *this;
+    if (m_encoder_model) {
+        cloned.m_encoder_model = m_encoder_model->clone();
+    } else {
+        if (m_encoder_request) {
+            cloned.m_encoder_request = m_encoder_request.get_compiled_model().create_infer_request();
+        }
+    }
```

Comment:
This implementation only handles the encoder, but `AutoencoderKL `also includes decoder (`m_decoder_model` / `m_decoder_request`)

Reply:
Good catch! Fixed it, cloning both now.